src/iosMain/kotlin/qrcode/render/QRCodeGraphics.ios.kt

```diff
@@ -236,3 +244,17 @@ actual open class QRCodeGraphics actual constructor(
         }
     }
 }
+
+/**
+ * Converts an [NSData] object to a [ByteArray].
+ * This function is useful when working in Kotlin/Native where data may be represented as [ByteArray]
+ */
+@OptIn(ExperimentalForeignApi::class)
+private fun NSData.toByteArray(): ByteArray {
+    val result = ByteArray(length.toInt())
+    if (result.isEmpty()) return result
+    result.usePinned {
+        memcpy(it.addressOf(0), bytes, length)
+    }
+    return result
+}
```

Comment:
Heya! First of all, thanks a lot for the contribution! It means a lot to me <3

As for this function, what do you thing about this minor "optimization"? _(mostly a slight rewrite to have a better readability)_

```suggestion
private fun NSData.toByteArray(): ByteArray {
    val arrayLen = length.toInt().coerceAtLeast(0)
    val result = ByteArray(arrayLen)
    
    return when (arrayLen) {
        0 -> result
        else -> result.usePinned { memcpy(it.addressOf(0), bytes, length) }
    }
}
```

Reply:
Hey, I'm glad I could give you a hand on this and contribute to improve a little bit more this awesome project.
Regarding the function, I'm totally okay with it and will address it asap.

Reply:
Oh, just checked and it does return but just like `let { }`

How about:

```kotlin
private fun NSData.toByteArray(): ByteArray {
    val arrayLen = length.toInt().coerceAtLeast(0)

    return ByteArray(arrayLen).apply {
        if (arrayLen > 0) {
            result.usePinned { memcpy(it.addressOf(0), bytes, length) }
        }
    }
}
```

I'm all for the most readable code possible, given readability doesn't take away from performance (and that's pretty much why I love Kotlin so much)

From your suggestions, I'd go with 2 _**if**_ `memCpy` is safe in the case `arrayLen <= 0` 

_P.S.: I'm not as well versed on Apple's C stuff, but I've seen some cases of "size = -1 means REALLY empty/'treat this as null' and not just `[]`" and that's where that `coerceAtLeast(0)` comes from :)_

Reply:
I'll go with your proposal then to be sure that this won't be a pain later.

Reply:
Thank you so much! I'll merge it and release a new version in a few hours <3